DynamicArrayHW/DynamicArray.cpp

```diff
@@ -0,0 +1,209 @@
+#include "DynamicArray.h"
+
+template<typename T>
+DynamicArray<T>::DynamicArray()
+{
+	size = 0;
+	capacity = 0;
+	array = new T[size];
+}
+
+template<typename T>
+inline DynamicArray<T>::DynamicArray(std::size_t size)
+{
+	this->size = size;
+	capacity = size;
+	array = new T[size];
+}
+
+template<typename T>
+DynamicArray<T>::DynamicArray(const DynamicArray& other)
+{
+	size = other.size;
+	capacity = other.capacity;
+	array = new T[size];
+	for (std::size_t i = 0; i < size; i++)
+	{
+		array[i] = other.array[i];
+	}
+}
```

Comment:
All template definitions should be in header, not cpp
That's is where your errors come from

Reply:
corrected